hudi-common/src/main/java/org/apache/hudi/common/table/read/HoodieBaseFileGroupRecordBuffer.java

```diff
@@ -79,25 +80,21 @@ public abstract class HoodieBaseFileGroupRecordBuffer<T> implements HoodieFileGr
   protected InternalSchema internalSchema;
   protected HoodieTableMetaClient hoodieTableMetaClient;
 
-  public HoodieBaseFileGroupRecordBuffer(HoodieReaderContext<T> readerContext,
-                                         HoodieTableMetaClient hoodieTableMetaClient,
-                                         Option<String> partitionNameOverrideOpt,
-                                         Option<String[]> partitionPathFieldOpt,
-                                         HoodieRecordMerger recordMerger,
-                                         TypedProperties props) {
+  public HoodieBaseFileGroupRecordBuffer(HoodieReaderContext<T> readerContext, FileGroupReaderState readerState) {
     this.readerContext = readerContext;
-    this.readerSchema = readerContext.getSchemaHandler().getRequiredSchema();
-    this.partitionNameOverrideOpt = partitionNameOverrideOpt;
-    this.partitionPathFieldOpt = partitionPathFieldOpt;
-    this.recordMergeMode = getRecordMergeMode(props);
-    this.recordMerger = recordMerger;
+    this.readerState = readerState;
+    this.readerSchema = readerState.getRequiredSchema();
+    this.partitionNameOverrideOpt = readerState.getPartitionNameOverrideOpt();
+    this.partitionPathFieldOpt = readerState.getPartitionPathFieldOpt();
+    this.recordMergeMode = readerState.getRecordMergeMode();
+    this.recordMerger = readerState.getRecordMerger();
```

Comment:
Can all of these member variables be removed and replaced by `readerState` calls instead?